src/Option.purs

```diff
@@ -1075,6 +1172,32 @@ get proxy option = (alter go proxy option).value
   go :: Data.Maybe.Maybe value -> Data.Maybe.Maybe value
   go value = value
 
+-- | Attempts to fetch all of the values from all of the keys of an option.
+-- |
+-- | If every key exists in the option, the record of values is returned in `Just _`.
+-- |
+-- | If any key does not exist in the option, `Nothing` is returned.
+-- |
+-- | E.g.
+-- | ```PureScript
+-- | someOption :: Option.Option ( foo :: Boolean, bar :: Int )
+-- | someOption = Option.insert (Data.Symbol.SProxy :: _ "bar") 31 Option.empty
+-- |
+-- | -- This will be `Nothing` because the key `foo` does not exist in the option.
+-- | bar :: Data.Maybe.Maybe (Record ( foo :: Boolean, bar :: Int))
+-- | bar = Option.getAll someOption
+-- |
+-- | -- This will be `Just { foo: true, bar: 31 }` because all keys exist in the option.
+-- | bar :: Data.Maybe.Maybe (Record ( foo :: Boolean, bar :: Int))
+-- | bar = Option.getAll (Option.insert (Data.Symbol.SProxy :: _ "foo") true someOption)
+-- | ```
+getAll ::
+  forall option record.
+  GetAll option record =>
+  Option option ->
+  Data.Maybe.Maybe (Record record)
+getAll = getAll'
```

Comment:
@bbarker is this what you're looking for? Lemme know if this is off-base, and we can try to change things for the better.

Reply:
@joneshf sorry for the long delay, many things happened I did not expect (2 car wrecks, and proposal writing, which hopefully isn't a wreck). I wish I'd just replied, immediately, because this does look like what I need.

Reply:
Yikes! Sorry to hear that. Hope everything worked out.

Reply:
All is well now, thanks! Though I haven't gotten back to writing a PureScript to MATLAB  transpiler since!